### PR TITLE
Issue34 bug prep data

### LINKED ISSUE #34: pspm_prepdata returns all NaNs when input signal has NaN value(s)
### Summary
`pspm_prepdata` cannot gracefully handle signals with NaN value(s). The returned signal contains all NaN values.

### Steps to Reproduce
1. Run the below code:
```MATLAB
filt.sr = 500;
filt.lpfreq = 20;
filt.lporder = 1;
filt.hpfreq = 5;
filt.hporder = 1;
filt.direction = 'uni';
filt.down = 200;
data = sin(1:10000);
data(10) = NaN;
[sts, outdata, newsr] = pspm_prepdata(data, filt);
```

### Expected Results
The function should somehow inputs with NaN values, or at least should give a warning about this issue.

### Actual Results
The function completes with success status, but `outdata` contains only NaN values.

### Technical Info
* PsPM revision ID: ea9490b
* MATLAB version: R2018b

### PR DESCRIPTION
**Pull request by irojkov-ph**
Fixes #34 .

Changes proposed in this pull request:
- Send a warning message("_Data contains NaN values._") if some NaN values are present in the data sent to the __prep_data__ function.
- Added a test line in __pspm_prepdata_test__
- Changed __Developper_Guide__

### CODE REVIEW

File: src/pspm_prepdata.m [hunk 1]
```diff
@@ -38,6 +38,12 @@
 newsr = 0;
 outdata = data;
 
+% check input for NaN values
+% -------------------------------------------------------------------------
+if any(isnan(data))
+    warning('ID:invalid_input', 'Data contains NaN values.');
+end
+
```
Comment (eozd) [inline src/pspm_prepdata.m]: Looks good. Could you also say something along the lines of prepdata can't handle nans, set the status to -1 and return immediately?

Comment (irojkov-ph) [reply src/pspm_prepdata.m]: is that what you mean ? (see the new commit)

Comment (eozd) [reply src/pspm_prepdata.m]: Exactly.

Comment (eozd) [reply src/pspm_prepdata.m]: I will approve after you remove the `.lyx~` file.

Comment (irojkov-ph) [reply src/pspm_prepdata.m]: Yeah sorry forgot it --'

**Pull request merged**